### DISCUSSION TITLE
CFG negative prompts

### DISCUSSION THREAD

**Discussion by raefu** [hlky/stable-diffusion-webui#999] [Feature suggestions]
This is distinct from negative prompt _weighting_, and is used directly by the classifier free guidance process.

Effectively, if you convert the line in webui.py from `uc = (model if not opt.optimized else modelCS).get_learned_conditioning(len(prompts) * [""])` to something like `uc = (model if not opt.optimized else modelCS).get_learned_conditioning(len(prompts) * ["blurry, bad anatomy, disfigured"])`, it will guide the generated image _away_ from images like the undesired negative prompt.

In the UI, this could be either part of the prompt, by extending the prompt weighting to treat negative weights as options for the _negative prompt_ (`human:1 bad anatomy:-1`), perhaps with a toggle, or another textbox.

Deeper discussion and an example (clothed anime woman): https://cdn.discordapp.com/attachments/1010980910063161406/1018369516905963640/unknown.png

**Comment by outlookhazy** [comment]
That's a very cool idea

**Comment by jangofett890** [comment]
This seems to be the only thing holding this web ui down, negative prompts generate such better pictures its not even worth all the extra features this repo has over Automatic1111's simply because of the lack of negative prompts. This needs to be added ASAP.

**Reply by jangofett890** [reply]
nvm, this thread should be closed, it was added with this repo change a few hours ago whoops https://github.com/sd-webui/stable-diffusion-webui/pull/1035
